clients/instance/ibm-pi-workspaces.go

```diff
@@ -54,3 +70,58 @@ func (f *IBMPIWorkspacesClient) GetAll() (*models.Workspaces, error) {
 	}
 	return resp.Payload, nil
 }
+
+// Create a workspace
+func (f *IBMPIWorkspacesClient) Create(name, location, groupID, plan string) (*resourcecontrollerv2.ResourceInstance, *core.DetailedResponse, error) {
+	resourceController, err := ibmpisession.CreateResourceControllerV2(f.session.Options.URL, f.session.Options.Authenticator)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error creating Resource Controller client: %v", err)
+	}
+	planID := translatePlan(plan)
+	if planID == "" {
+		return nil, nil, fmt.Errorf("workspace creation error, incorrect plan value; either \"public\" or \"private\" is allowed")
+	}
+	params := resourceController.NewCreateResourceInstanceOptions(name, location, groupID, planID)
+	controller, response, err := resourceController.CreateResourceInstance(params)
+	if err != nil {
+		return nil, response, fmt.Errorf("error creating workspace: controller %v response %v  err %v", controller, response, err)
+	}
+	if response.StatusCode >= 400 {
```

Comment:
Don't we also want to use `ibmpisession.SDKFailWithAPIError`?

Reply:
Possibly? I'm implementing calls from a different SDK, and I didn't want to introduce new potential errors. I would prefer to revise this after the break.